File: Java/AppGeradorQRCode/app/src/main/java/curso/kotlin/appgeradorqrcode/MainActivity.java
package curso.kotlin.appgeradorqrcode;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Bitmap;
import android.graphics.Color;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import com.google.zxing.BarcodeFormat;
import com.google.zxing.WriterException;
import com.google.zxing.common.BitMatrix;
import com.google.zxing.qrcode.QRCodeWriter;

public class MainActivity extends AppCompatActivity {
    
    EditText editQRCode;
    Button btnGerarQRCode;
    ImageView imageQRCode;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        
        
        initComponents();
        
        btnGerarQRCode.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(TextUtils.isEmpty(editQRCode.getText().toString())) {
                    editQRCode.setError("*");
                    editQRCode.requestFocus();
                } else {
                    gerarQRCode(editQRCode.getText().toString());
                }
            }
        });
    }

    private void initComponents() {
        editQRCode = findViewById(R.id.editQRCode);
        btnGerarQRCode = findViewById(R.id.btnGerarQRCode);
        imageQRCode = findViewById(R.id.imgQRCode);
    }

    private void gerarQRCode(String conteudoDoQRCode) {
        // zxing-android-embedded
        QRCodeWriter qrCode = new QRCodeWriter();

        try {
            BitMatrix bitMatrix = qrCode.encode(conteudoDoQRCode, BarcodeFormat.QR_CODE, 196, 196);
            
            int tamanho = bitMatrix.getWidth();
            int altura = bitMatrix.getHeight();
            
            Bitmap bitmap = Bitmap.createBitmap(tamanho, altura, Bitmap.Config.RGB_565);

            for (int i = 0; i < tamanho; i++) {
                for (int j = 0; j < altura; j++) {
                    bitmap.setPixel(i, j, bitMatrix.get(i,j) ? Color.BLACK : Color.WHITE);
                }
            }

            imageQRCode.setImageBitmap(bitmap);
            

        } catch(WriterException exception) {
            exception.printStackTrace();
        }
    }
}

File: Kotlin/AppGeradorQRCode/app/src/main/java/curso/kotlin/appgeradorqrcode/MainActivity.kt
package curso.kotlin.appgeradorqrcode

import android.graphics.Bitmap
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.TextUtils
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import androidx.core.graphics.get
import com.google.zxing.BarcodeFormat
import com.google.zxing.WriterException
import com.google.zxing.common.BitMatrix
import com.google.zxing.qrcode.QRCodeWriter

class MainActivity : AppCompatActivity() {

    var editQRCode: EditText? = null
    var btnGerarQRCode: Button? = null
    var imgQRCode: ImageView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initComponents();

        btnGerarQRCode!!.setOnClickListener {
            if(TextUtils.isEmpty(editQRCode!!.text.toString())) {
                editQRCode!!.error = "*"
                editQRCode!!.requestFocus()
            } else {
                gerarQRCode(editQRCode!!.text.toString())
            }
        }
    }

    fun gerarQRCode(conteudoQRCode: String) {
        // zxing-android-embedded

        val qrCode = QRCodeWriter()

        try {
            val bitMatrix = qrCode.encode(conteudoQRCode, BarcodeFormat.QR_CODE, 196, 196);

            val tamanho = bitMatrix.width
            val altura = bitMatrix.height

            val bitmap = Bitmap.createBitmap(tamanho, altura, Bitmap.Config.RGB_565)

            for (i in 0 until tamanho) {
                for (j in 0 until altura) {
                    bitmap.setPixel(i,j, if(bitMatrix.get(i,j)) Color.BLACK else Color.WHITE)
                }
            }

            imgQRCode!!.setImageBitmap(bitmap)

        } catch (exception: WriterException) {
            exception.printStackTrace()
        }
    }

    fun initComponents() {
        editQRCode = findViewById(R.id.editQRCode)
        btnGerarQRCode = findViewById(R.id.btnGerarQRCode)
        imgQRCode = findViewById(R.id.imgQRCode)
    }
}